src/upload.rs

```diff
@@ -32,6 +33,9 @@ pub enum UploadError {
     /// File already exists
     #[error("File already exists: {0}")]
     FileExistsError(String),
+    /// Read package metadata error
+    #[error("Read package metadata error")]
```

Comment:
How about "Could not read the metadata from the package at {0}" with the PathBuf as field in the error? Or simply "Could not read the metadata from the package" if the path is already included in the backtrace.